### PR TITLE
test(test): add social integration tests

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

add social integration tests

## Testing

test locally
@xiaoyijun cc: @logto-io/eng

### CODE REVIEW

**Review by wangsijie — APPROVED** [review]

File: packages/integration-tests/tests/social-session.test.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+    // User with social does not exist
+    expect(response instanceof HTTPError && response.response.statusCode === 422).toBe(true);
+
+    await signInWithUsernameAndPassword(username, password, client.interactionCookie);
+
+    await expect(
+      bindWithSocial(mockSocialConnectorId, client.interactionCookie)
+    ).resolves.not.toThrow();
+  });
```
Comment (xiaoyijun) [inline packages/integration-tests/tests/social-session.test.ts:120]: Maybe need to check if there are `identities` inside the user info data.

**Review by xiaoyijun — APPROVED** [review]
Overall LGTM

File: packages/integration-tests/tests/social-session.test.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+import MockClient from '@/client';
+import { setUpConnector, createUserByAdmin } from '@/helpers';
+import { generateUsername, generatePassword } from '@/utils';
+
+const state = 'foo_state';
+const redirectUri = 'http://foo.dev/callback';
+const code = 'auth_code_foo';
+
+describe('social sign-in and register', () => {
+  const userId = crypto.randomUUID();
```
Comment (IceHe) [inline packages/integration-tests/tests/social-session.test.ts:21]: The max length of `users.id` in the schema is 12.

Better align with the max length and extract a utility function `generateUserId` in `utils.ts`?

File: packages/integration-tests/tests/social-session.test.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      )
+    ).resolves.toBeTruthy();
+
+    const response = await getAuthWithSocial(
+      { connectorId: mockSocialConnectorId, data: { state, redirectUri, code, userId } },
+      client.interactionCookie
+    ).catch((error: unknown) => error);
+
+    // User with social does not exist
+    expect(response instanceof HTTPError && response.response.statusCode === 422).toBe(true);
```
Comment (IceHe) [inline packages/integration-tests/tests/social-session.test.ts:46]: Should be `.toBeTruthy()`?

File: packages/integration-tests/tests/social-session.test.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    );
+
+    await client.processSession(redirectTo);
+
+    expect(client.isAuthenticated).toBeTruthy();
+  });
+
+  /* Note: As currently we can not prepare a social identities through admin api.
+   * The sign-in test case MUST run concurrently after the register test case
+   */
```
Comment (IceHe) [inline packages/integration-tests/tests/social-session.test.ts:61]: Follow a better comment style?
```suggestion
  /*
   * Note: As currently we can not prepare a social identities through admin api.
   * The sign-in test case MUST run concurrently after the register test case
   */
```
See https://google.github.io/styleguide/jsguide.html#formatting-block-comment-style

File: packages/integration-tests/tests/social-session.test.ts (hunk 1 above)
Comment (simeng-li) [reply packages/integration-tests/tests/social-session.test.ts:120]: should throw if not included?

File: packages/integration-tests/tests/social-session.test.ts (hunk 2 above)
Comment (simeng-li) [reply packages/integration-tests/tests/social-session.test.ts:21]: it is used as social identity userId

File: packages/integration-tests/tests/social-session.test.ts (hunk 3 above)
Comment (simeng-li) [reply packages/integration-tests/tests/social-session.test.ts:46]: in this case could strictly equal to true as it is a bool type

File: packages/integration-tests/tests/social-session.test.ts (hunk 2 above)
Comment (IceHe) [reply packages/integration-tests/tests/social-session.test.ts:21]: Oh. It's the user ID from the social identity provider.

Maybe it should be named as `socialUserId`?

File: packages/integration-tests/tests/social-session.test.ts (hunk 4 above)
Comment (simeng-li) [reply packages/integration-tests/tests/social-session.test.ts]: updated

**Review by wangsijie — APPROVED** [review]

File: packages/integration-tests/tests/social-session.test.ts (hunk 1 above)
Comment (simeng-li) [reply packages/integration-tests/tests/social-session.test.ts:138]: updated

File: packages/integration-tests/tests/social-session.test.ts (hunk 2 above)
Comment (simeng-li) [reply packages/integration-tests/tests/social-session.test.ts]: updated

**Pull request merged**